editor/plugins/asset_library_editor_plugin.cpp

```diff
@@ -1432,6 +1440,39 @@ void EditorAssetLibrary::_update_asset_items_columns() {
 	asset_items_column_width = (get_size().x / new_columns) - (100 * EDSCALE);
 }
 
+void EditorAssetLibrary::_set_library_message(const String &p_message) {
+	library_message->set_text(p_message);
+
+	if (library_message_action.is_valid()) {
+		library_message_button->disconnect("pressed", library_message_action);
+		library_message_action = Callable();
+	}
+	library_message_button->hide();
+
+	library_message_box->show();
+}
+
+void EditorAssetLibrary::_set_library_message_with_action(const String &p_message, const String &p_action_text, const Callable &p_action) {
+	library_message->set_text(p_message);
+
+	library_message_button->set_text(p_action_text);
+	if (library_message_action.is_valid()) {
+		library_message_button->disconnect("pressed", library_message_action);
+		library_message_action = Callable();
+	}
+	library_message_action = p_action;
+	library_message_button->connect("pressed", library_message_action);
+	library_message_button->show();
+
+	library_message_box->show();
+}
+
+void EditorAssetLibrary::_force_online_mode() {
+	EditorSettings::get_singleton()->set_setting("network/connection/network_mode", EditorSettings::NETWORK_ONLINE);
+	EditorSettings::get_singleton()->notify_changes();
```

Comment:
Is the `notify_changes()` necessary? We use it only in the settings dialog (and there are more places where settings are changed).

Reply:
I think without that the notification isn't sent, and I rely on it elsewhere.

Reply:
I remember simply modifying the settings would cause propagation and saving automatically, but maybe it has changed.

Reply:
Supposedly doing `set`/`set_setting` should emit a signal which the editor settings dialog connects to and triggers the notification and the save on 1.5 second delay. I'll double check.

Reply:
Okay, so it works. But the delay makes it awkward. You click the button, and for a couple of seconds nothing happens. So I'll keep it for the buttons that are supposed to force the setting immediately.